addons/website/views/website_templates.xml

```diff
@@ -173,7 +173,7 @@
         <!-- website.is_public_user() is needed for menus having a page with
         restricted visibility (only shown to logged in user): public users and
         logged in users can't share the menu cache. -->
-        <attribute name="t-cache">xmlid,website,website.is_public_user()</attribute>
+        <attribute name="t-cache">xmlid,website,website.is_public_user() if website.has_record_url_in_menu() else None</attribute>
```

Comment:
That's part of the QWeb that is magic for me. Why when we pass `xmlid,website,website.is_public_user()`, it's interpreted as python and the function `is_public_user` is called but with your change `xmlid,website,website.is_public_user() if website.has_record_url_in_menu() else None` is not interpreted as python and so the functions are never called and the bug is not fixed. @rdeodoo ?